db_sql.go

```diff
@@ -133,31 +151,62 @@ func (x *sqlAuthenticationDB) ResetPasswordFinish(identity string, token string,
 	}
 }
 
-func (x *sqlAuthenticationDB) CreateIdentity(identity, password string) error {
+func (x *sqlUserStoreDB) CreateIdentity(identity, password string) error {
 	hash, ehash := computeAuthausHash(password)
 	if ehash != nil {
 		return ehash
 	}
 
-	if tx, etx := x.db.Begin(); etx == nil {
-		if _, ecreate := tx.Exec(`INSERT INTO authuser (identity, password) VALUES ($1, $2)`, identity, hash); ecreate == nil {
-			return tx.Commit()
+	//insert into user store
+	if tx1, etx1 := x.db.Begin(); etx1 == nil {
+		if _, ecreate1 := tx1.Exec(`INSERT INTO userstore (identity) VALUES ($1)`, identity); ecreate1 == nil {
+			tx1.Commit()
+
+			//get user id
+			var userId UserId
+			rows, err := x.db.Query(`SELECT userid FROM userstore WHERE identity = $1`, identity)
```

Comment:
Use `QueryRow` to make this code shorter.
This function must definitely be rewritten so that it's linear.
Also, you are starting transactions twice here. There is definitely no need for nested transactions here.